### ISSUE TITLE
False Positive for SolidJS REF declared with `let` rather than `const`

### ISSUE THREAD

**Issue by Ayehavgunne** [solidjs-community/eslint-plugin-solid#162]
**Describe the bug**
When defining a [ref](https://docs.solidjs.com/concepts/refs#refs-in-solid) I found out that the variable needs to be declared using `let` and not `const`. ESLint however want's the ref to be declared with `const` since it is not reassigned anywhere in my code.

**To Reproduce**
Define and use a ref in a component and then use ESLint on it.
Using TypeScript
`let input_ref: HTMLInputElement | undefined = undefined`

**Expected behavior**
ESLint needs to ignore this rule for refs to prevent a tricky to troubleshoot runtime error in SolidJS.

**Screenshots**

**Environment (please complete the following information):**

- OS: Mac OS 14.6.1 (23G93)
- Node version (`node --version`): v20.5.1
- `eslint-plugin-solid` version (`npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`): 0.14.3(eslint@9.11.1)(typescript@4.9.5)
- `eslint` version (`npm list eslint`/`yarn why eslint`): 9.11.1

**Comment by joshwilsonvu** [comment]
The error here is coming from the builtin rule `prefer-const`, so this plugin is unable to fix it. Best we could do is provide an alternative `solid/prefer-const` and instruct users to disable the builtin version.

Honestly, my recommendation is to use a signal as a ref, i.e. 

```tsx
const [input, setInput] = createSignal(null);
// ...
<input ref={setInput} // ...
```

Or, disable `prefer-const` if you plan on using `let` refs.

For now, given the complexity of [the builtin rule](https://github.com/eslint/eslint/blob/cceccc771631011e04b37122b990205f0e8b6925/lib/rules/prefer-const.js), I don't feel that it's worth the maintenance burden of copying and tweaking to provide ref-specific functionality. If this issue garners additional interest, I'll reconsider. Thanks for filing an issue!
